api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/convolutional/Conv3DTranspose.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2021-2022 JetBrains s.r.o. and Kotlin Deep Learning project contributors. All Rights Reserved.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlinx.dl.api.core.layer.convolutional
+
+import org.jetbrains.kotlinx.dl.api.core.activation.Activations
+import org.jetbrains.kotlinx.dl.api.core.initializer.HeNormal
+import org.jetbrains.kotlinx.dl.api.core.initializer.HeUniform
+import org.jetbrains.kotlinx.dl.api.core.initializer.Initializer
+import org.jetbrains.kotlinx.dl.api.core.layer.NoGradients
+import org.jetbrains.kotlinx.dl.api.core.layer.requireArraySize
+import org.jetbrains.kotlinx.dl.api.core.layer.toLongList
+import org.jetbrains.kotlinx.dl.api.core.regularizer.Regularizer
+import org.tensorflow.Operand
+import org.tensorflow.op.Ops
+import org.tensorflow.op.nn.Conv3dBackpropInput
+
+/**
+ * 3D convolution transpose layer.
+ *
+ * This is an operation going in the opposite direction of a normal convolution:
+ * it transforms a tensor shaped like an output of some convolution into tensor that has the shape of the input.
+ *
+ * This layer expects input data of size `(N, D, H, W, C)` where
+ * ```
+ * N - batch size
+ * D - depth
+ * H - height
+ * W - width
+ * C - number of channels
+ * ```
+ *
+ * Note: providing explicit output padding is currently not supported.
```

Comment:
Should we throw an exception on that or add a contract to `init `section?

Reply:
There is no `outputPadding` parameter, so there is no situation where we could throw an exception. The note is here since keras does have this parameter (see https://keras.io/api/layers/convolution_layers/convolution3d_transpose/).